plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergSystemTables.java

```diff
@@ -91,6 +97,28 @@ public void testPartitionTable()
         assertEquals(rowsByPartition.get(LocalDate.parse("2019-09-10")).getField(4), new MaterializedRow(DEFAULT_PRECISION, new MaterializedRow(DEFAULT_PRECISION, 4L, 5L, 0L)));
     }
 
+    @Test
+    public void testPartitionTableOnDropColumn()
+    {
+        MaterializedResult result = computeActual("SELECT * from test_schema.\"test_table_multi_column$partitions\"");
+
+        assertEquals(result.getRowCount(), 3);
+        Map<LocalDate, MaterializedRow> rowsByPartition = result.getMaterializedRows().stream()
+                .collect(toImmutableMap(row -> ((LocalDate) ((MaterializedRow) row.getField(0)).getField(0)), Function.identity()));
+
+        assertEquals(rowsByPartition.get(LocalDate.parse("2019-09-08")).getField(4), new MaterializedRow(DEFAULT_PRECISION,
+                new MaterializedRow(DEFAULT_PRECISION, "a", "a", 0L),
+                new MaterializedRow(DEFAULT_PRECISION, 0L, 0L, 0L)));
+
+        assertUpdate("ALTER TABLE IF EXISTS test_schema.test_table_multi_column drop column _varchar");
```

Comment:
This tests intend to show the results of before/after update.

Reply:
That's fine, I'm just saying the `CREATE TABLE` and `DROP TABLE` should be in the test method instead of in setup/tearDown.

Reason being, tables created in setup could be shared among multiple tests, and since this table is being altered the state of the table would depend on the order the tests are run.

Reply:
Bump on moving the create/drop table into the test.

You can also remove the `IF EXISTS` from the `ALTER TABLE`. If the table doesn't exist here we have a problem.